src/rpg_database/character_sheet/CharacterAttribute.java

```diff
@@ -48,9 +51,14 @@ public void setSelfValueByField(Fields field, Object value) {
 
 	@Override
 	public Object getStoredValueByField(Fields field) {
-		if (field.getAllowedClass() == Integer.class)
+		if (field.getAllowedClass() == Integer.class) {
+			if (field.equals(Fields.DEFENSE))
+				return defense + value;
+			else if (field.equals(Fields.SPEED))
+				return (getBackground().getBaseSpeed() + getArmorType().getArmorPenalty() + value) < 0 ? 0
+						: getBackground().getBaseSpeed() + getArmorType().getArmorPenalty() + value;
 			return value;
```

Comment:
miért van két return? Ez nem dead code? IDE nem jelezte?

Reply:
Nem, mert a felső beágyazott feltétel az csak arra vonatkozik, hogy ha def vagy speed-et kérsz ki. Az  utolsó return pedig az attributott adja vissza.

Reply:
A refaktorod rendben van, ami meg a kósza returnt illeti:

Áh értem. Viszont akkor ezt itt elég slendriánul csináltam meg. Lehet úgy lenne a legjobb ha lenne itt egy feltétel, ami az átadott fieldet vizsgálja, egyéb esetben pedig dob.... Mert így bármilyen Integer.class-ú field kikérheti a value-t....
pl. ha az ATTRIBUTES statikus mező egy Set vagy egy ArrayList lenne akkor azt faszán ki is lehetne erre használni Így:
```java
public final static ArrayList<Fields> ATTRIBUTES = new ArrayList<>( Arrays.asList(Fields.COMMUNICATION, Fields.CONSTITUTION, Fields.CUNNING, Fields.DEXTERITY, Fields.MAGIC,Fields.PERCEPTION, Fields.STRENGTH, Fields.WILLPOWER));
// Bla-bla-bla
//.
//.
//.
	} else if (field.equals(Fields.SPEED)) {
		int speedCalculation = getBackgroundBaseSpeed() + getArmorPenaltyValue() + value;
		return speedCalculation < 0 ? 0 : speedCalculation;
	} else if (ATTRIBUTES.contains(field)) {
		return value;
	}
} else if (field.getAllowedClass() == Boolean.class)
	return isMajor;
throw new InvalidParameterException(generateExceptionMessage(field));
```
Ha itt az utolsó sorban nem else-be kerül a dobás, hanem szimplán ott van a végén akkor pedig meg van oldva az is hogy dobjon abban az esetben, ha Integer.class-ú fieldet kap az 54-es sorban, de egyik belső feltételből sem tér vissza a függvény.

Ezt a korrekciót nem lenne kedved elvégezni? Esetleg egy külön commitban? (nyilván kell hozzá akkor egy tesztet is írni, ami megnézi hogy dob -e ha tilos Integer.class-ú fielddel próbálsz értéket kikérni)